src/Raven.Server/Documents/Handlers/Debugging/MemoryStatsHandler.cs

```diff
@@ -302,58 +308,132 @@ public static int LongestCommonPrefixLength(List<string> strings)
-            public long WorkingSet { get; set; }
-            public long TotalUnmanagedAllocations { get; set; }
-            public long ManagedAllocations { get; set;}
-            public long TotalMemoryMapped { get; set; }
-            public string PhysicalMem { get; set; }
-            public string FreeMem { get; set; }
-            public string HighMemLastOneMinute { get; set; }
-            public string LowMemLastOneMinute { get; set; }
-            public string HighMemLastFiveMinute { get; set; }        
-            public string LowMemLastFiveMinute { get; set; }
-            public string HighMemSinceStartup { get; set; }
-            public string LowMemSinceStartup { get; set; }
-            public MemoryInfoHumane Humane { get; set; }
-            public MemoryInfoMappingItem[] Mappings { get; set; }
-            //TODO: threads
-            
-        }
-        
-        internal class MemoryInfoHumane
-        {
-            public string WorkingSet { get; set; }
-            public string TotalUnmanagedAllocations { get; set; }
-            public string ManagedAllocations { get; set; }
-            public string TotalMemoryMapped { get; set; }
-        }
-        
-        internal class MemoryInfoMappingItem
-        {
-            public string Directory { get; set; }
-            public long TotalDirectorySize { get; set; }
-            public string HumaneTotalDirectorySize { get; set; }
-            public Dictionary<string, MemoryInfoMappingFileInfo> Details { get; set; }
-        }
-        
-        internal class MemoryInfoMappingFileInfo
+
+        [RavenAction("/admin/debug/proc/meminfo", "GET", AuthorizationStatus.DatabaseAdmin, IsDebugInformationEndpoint = true)]
+        public Task ProcMemInfo()
         {
-            public long FileSize { get; set; }
-            public string HumaneFileSize { get; set; }
-            public long TotalMapped { get; set; }
-            public string HumaneTotalMapped { get; set; }
-            public MemoryInfoMappingDetails[] Mappings { get; set; }
+            var txt = File.ReadAllLines("/proc/meminfo");
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                var djv = new DynamicJsonValue();
+                var djvHumanly = new DynamicJsonValue();
+                foreach (var line in txt)
+                {
+                    var parsedLine = line.Split(new[] {' '}, StringSplitOptions.RemoveEmptyEntries);
+
+                    if (parsedLine.Length != 3) // format should be: {filter}: <num> kb
+                        continue; // an error but we ignore and do not put in the results
+
+                    var key = parsedLine[0].Split(":")[0];
+
+                    if (long.TryParse(parsedLine[1], out var result) == false)
+                    {
+                        djv[key] = "Failed To Parse Value : " + parsedLine[1];
+                        continue;
+                    }
```

Comment:
For me this piece of code is fragile. Maybe let's do it this way but also include the raw line. Also if we get exception here we'll still be able to output raw message.